service/resource/configmap/current.go

```diff
@@ -0,0 +1,32 @@
+package configmap
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/microerror"
+)
+
+// GetCurrentState returns the current state of the prometheus config configmap.
+// If the configmap exists, it is returned.
+// If the configmap does not exist, nil is returned.
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	r.logger.Log("debug", fmt.Sprintf("fetching configmap: %s/%s", r.configMapNamespace, r.configMapName))
+
+	configMap, err := r.k8sClient.CoreV1().ConfigMaps(r.configMapNamespace).Get(
+		r.configMapName, metav1.GetOptions{},
+	)
+	if errors.IsNotFound(err) {
+		r.logger.Log("debug", "configmap does not exist")
+		return nil, nil
+	} else if err != nil {
+		r.logger.Log("error", "an error occurred fetching configmap", "message", err.Error())
```

Comment:
The error is already logged down the line. What would be better would be to annotate the error using `Maskf` and remove the logging here.

Reply:
Yeah, I get you. Updating.